src/test/test-notify-recv.c

```diff
@@ -80,7 +91,17 @@ TEST(notify_socket_prepare) {
         if (r == 0) {
                 ASSERT_OK_ERRNO(setenv("NOTIFY_SOCKET", path, /* overwrite = */ true));
                 ASSERT_OK_POSITIVE(sd_notify(/* unset_environment = */ false, "FIRST_MESSAGE=1"));
-                ASSERT_OK_POSITIVE(sd_notify(/* unset_environment = */ false, "FIRST_MESSAGE=2\nADDITIONAL_DATA=hoge"));
+
+                _cleanup_close_ int fd1 = open("/tmp", O_RDONLY|O_CLOEXEC|O_DIRECTORY);
+                _cleanup_close_ int fd2 = open("/dev/null", O_RDONLY|O_CLOEXEC);
+                ASSERT_TRUE(fd1 >= 0);
+                ASSERT_TRUE(fd2 >= 0);
```

Comment:
Please check error one by one with `ASSERT_OK_ERRNO()`, otherwise, errno may be overwritten by the second open():
```c
_cleanup_close_ int fd1 = open(...);
ASSERT_OK_ERRNO(fd1);
_cleanup_close_ int fd2 = open(...);
ASSERT_OK_ERRNO(fd2);
```